Automattic/wp-calypso
# Dotcom FSE Beta: Show opt-in step to 1% of existing users

Based on https://github.com/Automattic/wp-calypso/pull/56514.

#### Changes proposed in this Pull Request

* Show the Dotcom FSE Beta opt-in step to 1% of existing users.
* Also keep showing the step on environments with `full-site-editing/beta-opt-in` enabled.

The percentage is determined on the user ID with a simple formula, which lends itself to being quickly adjusted at will.
```js
userId % 100 < percentage
```

#### Note

There already is a [different implementation](https://github.com/Automattic/wp-calypso/blob/ce4a877e24ae50f019eb9b60156e69ebcdbad6f8/client/lib/naive-client-side-rollout/index.ts) of this "naive rollout" in Calypso.
I've asked the relevant team about it, and if it would suit our needs.

I've tried it briefly, and I've noticed that the `currentUser` object used by it is always undefined. I'll investigate it if we think reusing that function is better.

Although, I think that my super simple implementation is enough for the rollout, and it also gives us some degree of (manual) control over who gets to be eligible to the beta.

#### Testing instructions



* Make note of your user ID.
* Spin up local Calypso.
* Open http://calypso.localhost:3000/new with an existing user.
  * Make sure the second step is the FSE Beta opt-in.
* Open http://calypso.localhost:3000/new?flags=-full-site-editing/beta-opt-in (to disable the feature flag) with an exiting user.
  * If your user ID's last 2 digits are `< 10`:
    * Make sure the second step is the FSE Beta opt-in.
  * Otherwise:
    * Make sure the FSE Beta opt-in step is never shown.
    * Adjust the `FSE_BETA_ROLLOUT_PERCENTAGE` to be strictly `>` your user ID's last 2 digits.
    (e.g. user `12345` would need a percentage of `46` to be eligible to the FSE Beta)
    * Try again, and make sure this time the second step is the FSE Beta opt-in.
* Open http://calypso.localhost:3000/new in an anonymous tab (or logged out).
  * Progress through the steps.
  * Make sure the FSE Beta opt-in step is never shown.



Fixes #56506
